src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -563,10 +543,7 @@ export function makeItem(
   const tier = itemDef.inventory ? defs.ItemTierType[itemDef.inventory.tierTypeHash] : null;
   createdItem.infusionProcess = tier && tier.infusionProcess;
   createdItem.infusionFuel = Boolean(
-    createdItem.infusionProcess &&
-      itemDef.quality &&
-      itemDef.quality.infusionCategoryHashes &&
-      itemDef.quality.infusionCategoryHashes.length
+    createdItem.infusionProcess && idx(itemDef, (i) => i.quality.infusionCategoryHashes.length)
```

Comment:
Slight nitpick, but when I checked the generated babel output, it'll generate something equivalent to this:

```js
itemDef && 
itemDef.quality &&	
itemDef.quality.infusionCategoryHashes &&	
itemDef.quality.infusionCategoryHashes.length
```

If you know `itemDef` isn't undefined, you could write this as `idx(itemDef.quality, (q) => q.infusionCategoryHashes.length)` and it'd be more equivalent to the original code.

Reply:
reviewed against previous and corrected some idx instances to start with known values